### ISSUE TITLE
[BUG] Huge amount of snapshots that can't be deleted 1.4.3

### ISSUE THREAD

**Issue by simonebenati** [longhorn/longhorn#8284]
## Describe the bug

I have so many snapshots that can't be deleted automatically

## To Reproduce

Run a backup feature on volumes used by prometheus or elasticsearch

## Expected behavior

Not having so many snapshots taking up so much space

## Support bundle for troubleshooting

-

## Environment

 - Longhorn version: 1.4.3
 - Impacted volume (PV): 
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: rke2 1.26.11
   - Number of control plane nodes in the cluster:
   - Number of worker nodes in the cluster:
 - Node config
   - OS type and version:
   - Kernel version:
   - CPU per node:
   - Memory per node:
   - Disk type (e.g. SSD/NVMe/HDD):
   - Network bandwidth between the nodes (Gbps):
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster:

## Additional context

Could you please provide a proper script to delete snapshots I tried using one of mine that runs `kubectl delete snapshots.longhorn.io <snap name>` and then kubectl patch to remove the finalizer but the volume size doesn't go down. or reference a proper guide on how to setup these jobs properly?
Now if I run the recurring job to delete snapshots it won't delete the system hidden, if i run the snapshot cleanup it won't delete the user created snapshots. so i'm in a loop.

**Comment by simonebenati** [comment]
Also here, I checked many other issues but without any luck to solve my issue.

**Comment by ChanYiLin** [comment]
Hi @simonebenati 
you can try to create a recurring job to periodically clean up the snapshot
ref: https://longhorn.io/docs/1.6.1/snapshots-and-backups/scheduling-backups-and-snapshots/
Thanks

**Comment by ChanYiLin** [comment]
And if you are using recurring job to do the backup work
please make sure you have enabled this setting
https://longhorn.io/docs/1.6.1/references/settings/#automatically-clean-up-outdated-snapshots-of-recurring-backup-jobs

**Comment by simonebenati** [comment]
Hello, @ChanYiLin 

I am on 1.4.3 and don't have the the 'cleanup outdated snapshots' option.

Is it okay to run weekly a snapshot cleanup recurring job, while running daily backups against volumes?
Will it create issues with the backup stored on s3? Could it delete some backups on s3, by deleting corresponding snapshots? (I don't think so)

**Comment by ChanYiLin** [comment]
No, backup and snapshot are not dependent after they are created
Deleting the snapshot won't lead to the deletion of the backup.

> Is it okay to run weekly a snapshot cleanup recurring job, while running daily backups against volumes?
Yes, looks good to me.

**Comment by simonebenati** [comment]
@ChanYiLin Thank you very much for the support :D
